diff -ruN JumpTest-Step4/JumpTest.java JumpTest-Step5/JumpTest.java
--- JumpTest-Step4/JumpTest.java	2014-07-10 16:14:34.176628393 -0400
+++ JumpTest-Step5/JumpTest.java	2014-07-10 16:19:43.626730091 -0400
@@ -3,6 +3,7 @@
 import java.awt.Graphics;
 import java.awt.Color;
 import java.awt.Toolkit;
+import java.awt.event.KeyEvent;
 
 import javax.swing.Timer;
 
@@ -31,13 +32,19 @@
         Toolkit.getDefaultToolkit().sync();
     }
 
+    public void pressKey (int code) {
+        if (code == KeyEvent.VK_ESCAPE) {
+            System.exit (0);
+        }
+    }
+
     public void update () {
         player.update ();
     }
 
     public JumpTest() {
         this.addWindowListener(new JumpTestWindowAdapter());
-        this.addKeyListener(new JumpTestKeyAdapter());
+        this.addKeyListener(new JumpTestKeyAdapter(this));
         this.setUndecorated(true);
         this.setSize(320, 240);
         this.setVisible(true);
diff -ruN JumpTest-Step4/JumpTestKeyAdapter.java JumpTest-Step5/JumpTestKeyAdapter.java
--- JumpTest-Step4/JumpTestKeyAdapter.java	2014-07-10 16:14:06.728441972 -0400
+++ JumpTest-Step5/JumpTestKeyAdapter.java	2014-07-10 16:19:54.734805534 -0400
@@ -3,12 +3,18 @@
 
 public class JumpTestKeyAdapter extends KeyAdapter {
 
+    private JumpTest jt;
+
+    public JumpTestKeyAdapter (JumpTest jt) {
+        this.jt = jt;
+    }
+
     public void keyReleased (KeyEvent e) {
         System.out.println("You released key " + e.getKeyCode());
     }
 
     public void keyPressed (KeyEvent e) {
-        System.out.println("You pressed key " + e.getKeyCode());
+        jt.pressKey (e.getKeyCode ());
     }
 
 }
