diff -ruN AnimTest-Step2/AnimTest.java AnimTest-Step3/AnimTest.java
--- AnimTest-Step2/AnimTest.java	2014-07-07 13:07:39.234949807 -0400
+++ AnimTest-Step3/AnimTest.java	2014-07-07 13:11:25.224476550 -0400
@@ -1,5 +1,7 @@
 import java.awt.Frame;
 
+import javax.swing.Timer;
+
 public class AnimTest extends Frame {
 
     public AnimTest() {
@@ -7,6 +9,9 @@
         this.setUndecorated(true);
         this.setSize(320, 240);
         this.setVisible(true);
+
+        Timer t = new Timer(1000, new AnimTestActionListener());
+        t.start();
     }
 
     public static void main (String[] args) {
diff -ruN AnimTest-Step2/AnimTestActionListener.java AnimTest-Step3/AnimTestActionListener.java
--- AnimTest-Step2/AnimTestActionListener.java	1969-12-31 19:00:00.000000000 -0500
+++ AnimTest-Step3/AnimTestActionListener.java	2014-07-07 13:10:59.424302249 -0400
@@ -0,0 +1,9 @@
+import java.awt.event.ActionListener;
+import java.awt.event.ActionEvent;
+
+public class AnimTestActionListener implements ActionListener {
+
+    public void actionPerformed (ActionEvent e) {
+        System.out.println("Doing action");
+    }
+}
